pandas/io/common.py

```diff
@@ -314,11 +328,17 @@ def _get_handle(path_or_buf, mode, encoding=None, compression=None,
     handles : list of file-like objects
         A list of file-like object that were openned in this function.
     """
+    need_text_wrapping = [BytesIO]
     try:
         from s3fs import S3File
-        need_text_wrapping = (BytesIO, S3File)
+        need_text_wrapping.append(S3File)
+    except ImportError:
+        pass
+    try:
+        from hdfs3 import HDFile
+        need_text_wrapping.append(HDFile)
     except ImportError:
-        need_text_wrapping = (BytesIO,)
+        pass
 
```

Comment:
you need

``need_text_wrapping = tuple(need_text_wrapping)`` before ``handles``